ERP Spec revisions

adieuadieu opened the pull request:
Initial changes to the spec as we (@davgun29 & @chrisallthings) discussed in our meeting 2018-06-12.

Items which require follow-up:

- [x] checkIns.csv has been renamed
- [x] checkIns.csv's `tenantId` field is no longer valid. E.g. we also need registrationCodeId and companyId, or a way to simplify this
- [ ] what are all the fields for the tenants.csv?
- [ ] what are all the fields for the companies.csv?
- [x] propose renaming `manifest.json` to something else like `settings.json` while we're at it.

Feel free to add/make additional changes to this branch. Also remember that this thread is publicly visible.

<img width="1360" alt="screen shot 2018-06-12 at 2 18 38 pm" src="https://user-images.githubusercontent.com/438848/41289941-9d88e230-6e4b-11e8-95ca-b51f6c2a846e.png">

adieuadieu committed:
initial draft of changes to extend spec for emails, companies

Items which require follow-up:

- [ ] checkIns.csv has been renamed
- [ ] checkIns.csv's `tenantId` field is no longer valid. E.g. we also need registrationCodeId and companyId, or a way to simplify this
- [ ] what are all the fields for the tenants.csv?
- [ ] what are all the fields for the companies.csv?
- [ ] propose renaming manifest.json to something else like settings.json while we're at it.

adieuadieu committed:
fix typo

chrisallthings commented on erp-data-exchange-specification.md:
```diff
@@ -12,10 +12,12 @@ This document specifies the Allthings ERP Data Exchange format.
+    1.  [registrationCodes.csv](#registrationcodescsv)
     1.  [tenants.csv](#tenantscsv)
+    1.  [companies.csv](#companiescsv)
     1.  [propertyTeams.csv](#propertyteamscsv)
```
Should be renamed to checkIns.csv, right?

immerwiederchia commented on erp-data-exchange-specification.md:
```diff
@@ -12,10 +12,12 @@ This document specifies the Allthings ERP Data Exchange format.
     1.  [units.csv](#unitscsv)
     1.  [utilisationPeriods.csv](#utilisationperiodscsv)
     1.  [tenantCheckIns.csv](#tenantcheckinscsv)
```
this should be renamed into checkIns.csv and linked to the right capter

immerwiederchia commented on erp-data-exchange-specification.md:
```diff
@@ -28,15 +30,15 @@ This document specifies the Allthings ERP Data Exchange format.
+There are 11 recognised CSV files:
 [_uuidRemappings.csv_](#uuidremappingscsv), [_properties.csv_](#propertiescsv), [_groups.csv_](#groupscsv), [_units.csv_](#unitscsv),
-[_utilisationPeriods.csv_](#utilisationperiodscsv), [_tenantCheckIns.csv_](#tenantcheckinscsv), [_tenants.csv_](#tenantscsv),
+[_utilisationPeriods.csv_](#utilisationperiodscsv), [_tenantCheckIns.csv_](#tenantcheckinscsv), [registrationCodes.csv_](#registrationcodescsv), [_tenants.csv_](#tenantscsv), [companies.csv_](#companies),
```
also here the tenantCheckIns.csv should be renamed in to checkIns.csv

immerwiederchia committed:
Update erp-data-exchange-specification.md

immerwiederchia committed:
Update erp-data-exchange-specification.md

immerwiederchia committed:
Update erp-data-exchange-specification.md

immerwiederchia committed:
Update erp-data-exchange-specification.md

immerwiederchia committed:
Update erp-data-exchange-specification.md

immerwiederchia committed:
Update erp-data-exchange-specification.md

immerwiederchia committed:
Update erp-data-exchange-specification.md

immerwiederchia committed:
Update erp-data-exchange-specification.md

immerwiederchia reviewed (commented):
Great thx. I think this looks good for what we know now. I will go through the details now.

immerwiederchia commented:
Hi Marco, 

I agree with all of your comments. From my point of view the biggest questions are those two: 
- [ ] what are all the fields for the tenants.csv?
- [ ] what are all the fields for the companies.csv?

But we can't answer them yet and we have to wait for team orange for the final concept of the company data modell.

immerwiederchia committed:
Update erp-data-exchange-specification.md

immerwiederchia committed:
Update erp-data-exchange-specification.md

immerwiederchia committed:
Update erp-data-exchange-specification.md

immerwiederchia committed:
Update erp-data-exchange-specification.md

immerwiederchia commented:
@adieuadieu I just realized that i missed one point of your comment.  

- [x] checkIns.csv's `tenantId` field is no longer valid. E.g. we also need registrationCodeId and companyId, or a way to simplify this

I checked this and did some changes to the documentation (see checkIns.csv and registrationCodes.csv) I think this should work now. Right? 

RegistrationCodes would be optional in this model but if one wants to use a registrationCode for serveral Units and let's say one company. In the checkIns.csv all the fields are required - if not we wouldn't know what company can use what registrationCode. Right? 
![checkins](https://user-images.githubusercontent.com/38314404/43712325-0e660c26-9976-11e8-980e-fd715379fd9b.PNG)

iamn1ck commented on erp-data-exchange-specification.md:
````diff
@@ -154,63 +156,112 @@ The `utilisationPeriods.csv` describes utilisation periods.
+#### Example
+
+```csv
+importType,residentId,name
````
missing a column here

immerwiederchia committed:
Update erp-data-exchange-specification.md

adashst commented on erp-data-exchange-specification.md:
```diff
@@ -154,63 +156,111 @@ The `utilisationPeriods.csv` describes utilisation periods.
+
+### registrationCodes.csv
+
+The `registrationCodes.csv` file describes registration codes for one or more utilizationPeriods. It contains the registrationCode and the reseidentId of those entities that are checked into these utilizationPeriods.
```
small Typo: reseidentId -> residentId

adashst commented on erp-data-exchange-specification.md:
```diff
@@ -154,63 +156,111 @@ The `utilisationPeriods.csv` describes utilisation periods.
 | -------------------- | --------------------------- | ------------------------ |
 | **importType**       | [Import Type](#import-type) | Must always use "insert"
-| **id**               | [UUID](#uuid-type)          | Your UUID for the tenant |
+| **id**               | [UUID](#uuid-type)          | UUID to identify registrationCode |
```
1) shouldn't id be: registrationCodeId
2) residentId is missing

adashst commented on erp-data-exchange-specification.md:
````diff
@@ -154,63 +156,111 @@ The `utilisationPeriods.csv` describes utilisation periods.
 
 ```csv
-importType,id,registrationCode
+importType,id,residentId,registrationCode
````
id -> registrationCodeId

adashst commented on erp-data-exchange-specification.md:
```diff
@@ -154,63 +156,111 @@ The `utilisationPeriods.csv` describes utilisation periods.
+### tenants.csv
 
-The `tenantCheckIns.csv` file describes the relation between a tenant and a utilisation period.
+The `tenants.csv` file describes a tenant.
```
Shouldn't we be consistent in terms of why is the file called `tenants.csv` when it contains residentIds?

adashst commented on erp-data-exchange-specification.md:
```diff
@@ -154,63 +156,111 @@ The `utilisationPeriods.csv` describes utilisation periods.
+| Field                | Type                        | Description              |
+| -------------------- | --------------------------- | ------------------------ |
+| **importType**       | [Import Type](#import-type) | One of:<br/>`insert`, `update` 
+| **id**               | [UUID](#uuid-type)          | Your UUID for the company |
```
id -> companyId?

adashst commented on erp-data-exchange-specification.md:
```diff
@@ -154,63 +156,111 @@ The `utilisationPeriods.csv` describes utilisation periods.
-insert,7d87a383-1778-401b-a421-b60c900479c3,db8b732f-e0ff-41d9-9c15-ca1be2776fd4
-insert,7d87a383-1778-401b-a421-b60c900479c3,ad257d42-1078-4279-9918-e774859555ae
-insert,7d87a383-1778-401b-a421-b60c900479c3,ad257d42-1078-4279-9918-e774859555ae
+importType,residentId,name
```
importType,companyId,name,email

adashst commented on erp-data-exchange-specification.md:
```diff
@@ -154,63 +156,111 @@ The `utilisationPeriods.csv` describes utilisation periods.
+insert,db8b732f-e0ff-41d9-9c15-ca1be2776fd4,Big Company, Inc., info@bigcomp.com
+insert,ad257d42-1078-4279-9918-e774859555ae,Auto GmbH, ceo@autogmbh.com
+insert,2f0fc4c6-3a5e-4cd7-9007-9add50653be5,Fuji LLC, boss@fuji.com
+insert,41f93044-e050-4a33-9e52-ebdada55f9a7,West & Smith, Inc., manager@ws.com
```
Strings that include commas should be enclosed in " 
like:
insert,41f93044-e050-4a33-9e52-ebdada55f9a7,"West & Smith, Inc.", manager@ws.com

Commit:
Update erp-data-exchange-specification.md

Fixed some typos; Added some missing columns; Updated some ids to be more meaningful.

adashst reviewed (commented):
Did some changes though not being 100% sure about how the structure should be.
